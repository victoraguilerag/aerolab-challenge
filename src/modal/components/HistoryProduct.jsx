import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Loader from 'react-loader-spinner'
import coin from '../../icons/coin.svg'

export class HistoryProduct extends Component {
  state = {
    loading: true
  }
  showImage = (e) => {
    this.setState({
      loading: false
    })
  }
  render () {
    const {
      name,
      category,
      img,
      cost
    } = this.props.info
    return (
      <div className="product-card" style={styles.productCard} >
        {
          this.state.loading &&
            <div className="loader" style={styles.loader}>
              <Loader
                type="TailSpin"
                color="#00bfff"
                height="50"
                width="100"
              />
            </div>
        }
        <img
          src={img.url}
          srcSet={`${img.hdUrl} 2x`}
          alt={name}
          style={styles.productImage}
          onLoad={(e) => this.showImage(e)}
          onError={() => this.showImage()}
        />
        <div className="productUnavailable" style={styles.productUnavailable}>
          <p className="productUnavailableText" style={styles.productUnavailableText}>{`${cost}`}</p>
          <img
            className="productUnavailableCoin"
            style={styles.productUnavailableCoin}
            width="24px"
            height="24px"
            src={coin}
            alt="coin"
          />
        </div>
        <div className="product-info" style={styles.productInfo}>
          <div className="product-category" style={styles.productCategory}>
            {category}
          </div>
          <div className="product-name" style={styles.productName}>
            {name}
          </div>
        </div>
      </div>
    )
  }
}

const styles = {
  'productCard': {
    'background': '#ffffff',
    'boxShadow': '2px 2px 4px 0 rgba(0,0,0,0.10)',
    'width': '276px',
    'height': '276px',
    'position': 'relative',
    'cursor': 'pointer'
  },
  'productImage': {
    'background': '#ffffff',
    'width': '252px',
    'height': '182px',
    'marginTop': '12px',
    'pointerEvents': 'none'
  },
  'productInfo': {
    'position': 'absolute',
    'bottom': '15px',
    'left': '24px',
    'pointerEvents': 'none'
  },
  'productCategory': {
    'fontSize': '16px',
    'color': '#a3a3a3',
    'letterSpacing': '-0.04px',
    'textAlign': 'left',
    'pointerEvents': 'none'
  },
  'productName': {
    'fontSize': '18px',
    'color': '#616161',
    'letterSpacing': '-0.04px',
    'textAlign': 'left',
    'pointerEvents': 'none'
  },
  'productUnavailable': {
    'position': 'absolute',
    'top': '12px',
    'right': '12px',
    'pointerEvents': 'none',
    'display': 'flex',
    'justifyContent': 'center',
    'alignItems': 'center',
    'opacity': '0.8',
    'background': '#616161',
    'borderRadius': '100px',
    'width': '142px',
    'height': '42px'
  },
  'productUnavailableText': {
    'fontSize': '14px',
    'color': '#ffffff',
    'letterSpacing': '-0.03px',
    'textAlign': 'right'
  },
  'productUnavailableCoin': {
    'width': '20px',
    'height': '20px',
    'margin': '0 2.5px'
  }
}

HistoryProduct.propTypes = {
  info: PropTypes.shape({
    name: PropTypes.string.isRequired,
    category: PropTypes.string.isRequired,
    img: PropTypes.shape({
      url: PropTypes.string.isRequired,
      hdUrl: PropTypes.string.isRequired
    }),
    cost: PropTypes.number.isRequired
  })
}

export default HistoryProduct
