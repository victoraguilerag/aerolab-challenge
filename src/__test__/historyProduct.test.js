import React from 'react'
import renderer from 'react-test-renderer'
import { fromJS } from 'immutable'
import { HistoryProduct } from '../modal/components/HistoryProduct.jsx'
import productsStub from '../stub/products.js'

const coins = 2000
const selected = true
const info = fromJS(productsStub[0])
const functionMock = () => console.log('mock')

test('Hitory Product renders', () => {
  const component = renderer.create(
    <HistoryProduct
      coins={coins}
      selected={selected}
      info={info}
      handleRedeemProduct={functionMock}
      handleModal={functionMock}
      handleProductClick={functionMock}
    />,
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
