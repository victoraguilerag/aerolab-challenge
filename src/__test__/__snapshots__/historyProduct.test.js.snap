// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hitory Product renders 1`] = `
<div
  className="product-card"
  style={
    Object {
      "background": "#ffffff",
      "boxShadow": "2px 2px 4px 0 rgba(0,0,0,0.10)",
      "cursor": "pointer",
      "height": "276px",
      "position": "relative",
      "width": "276px",
    }
  }
>
  <div
    className="loader"
  >
    <div>
      <svg
        height="50"
        viewBox="0 0 38 38"
        width="100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <linearGradient
            id="a"
            x1="8.042%"
            x2="65.682%"
            y1="0%"
            y2="23.865%"
          >
            <stop
              offset="0%"
              stopColor="#00bfff"
              stopOpacity="0"
            />
            <stop
              offset="63.146%"
              stopColor="#00bfff"
              stopOpacity=".631"
            />
            <stop
              offset="100%"
              stopColor="#00bfff"
            />
          </linearGradient>
        </defs>
        <g
          fill="none"
          fillRule="evenodd"
        >
          <g
            transform="translate(1 1)"
          >
            <path
              d="M36 18c0-9.94-8.06-18-18-18"
              id="Oval-2"
              stroke="url(#a)"
              strokeWidth="2"
            >
              <animateTransform
                attributeName="transform"
                dur="0.9s"
                from="0 18 18"
                repeatCount="indefinite"
                to="360 18 18"
                type="rotate"
              />
            </path>
            <circle
              cx="36"
              cy="18"
              fill="#fff"
              r="1"
            >
              <animateTransform
                attributeName="transform"
                dur="0.9s"
                from="0 18 18"
                repeatCount="indefinite"
                to="360 18 18"
                type="rotate"
              />
            </circle>
          </g>
        </g>
      </svg>
    </div>
  </div>
  <img
    alt="Secador de Cabello"
    onError={[Function]}
    onLoad={[Function]}
    srcSet="undefined 2x"
    style={
      Object {
        "background": "#ffffff",
        "height": "182px",
        "marginTop": "12px",
        "pointerEvents": "none",
        "width": "252px",
      }
    }
  />
  <div
    className="productUnavailable"
    style={
      Object {
        "alignItems": "center",
        "background": "#616161",
        "borderRadius": "100px",
        "display": "flex",
        "height": "42px",
        "justifyContent": "center",
        "opacity": "0.8",
        "pointerEvents": "none",
        "position": "absolute",
        "right": "12px",
        "top": "12px",
        "width": "142px",
      }
    }
  >
    <p
      className="productUnavailableText"
      style={
        Object {
          "color": "#ffffff",
          "fontSize": "14px",
          "letterSpacing": "-0.03px",
          "textAlign": "right",
        }
      }
    >
      2000
    </p>
    <img
      alt="coin"
      className="productUnavailableCoin"
      height="24px"
      src={Object {}}
      style={
        Object {
          "height": "20px",
          "margin": "0 2.5px",
          "width": "20px",
        }
      }
      width="24px"
    />
  </div>
  <div
    className="product-info"
    style={
      Object {
        "bottom": "15px",
        "left": "24px",
        "pointerEvents": "none",
        "position": "absolute",
      }
    }
  >
    <div
      className="product-category"
      style={
        Object {
          "color": "#a3a3a3",
          "fontSize": "16px",
          "letterSpacing": "-0.04px",
          "pointerEvents": "none",
          "textAlign": "left",
        }
      }
    >
      electrodomestico
    </div>
    <div
      className="product-name"
      style={
        Object {
          "color": "#616161",
          "fontSize": "18px",
          "letterSpacing": "-0.04px",
          "pointerEvents": "none",
          "textAlign": "left",
        }
      }
    >
      Secador de Cabello
    </div>
  </div>
</div>
`;
